enterprise/coderd/templates.go

```diff
@@ -338,3 +341,45 @@ func (api *API) RequireFeatureMW(feat codersdk.FeatureName) func(http.Handler) h
 		})
 	}
 }
+
+// @Summary Invalidate prebuilt workspaces for template
+// @ID invalidate-prebuilt-workspaces-for-template
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Enterprise
+// @Param template path string true "Template ID" format(uuid)
+// @Success 200 {object} codersdk.InvalidatePrebuildsResponse
+// @Router /templates/{template}/prebuilds/invalidate [post]
+func (api *API) postInvalidateTemplatePrebuilds(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	template := httpmw.TemplateParam(r)
+
+	// Authorization: user must be able to update the template
+	if !api.Authorize(r, policy.ActionUpdate, template) {
+		httpapi.ResourceNotFound(rw)
+		return
+	}
+
+	// Update last_invalidated_at for all presets of the active template version
+	invalidatedPresets, err := api.Database.UpdatePresetsLastInvalidatedAt(ctx, database.UpdatePresetsLastInvalidatedAtParams{
+		TemplateID:        template.ID,
+		LastInvalidatedAt: sql.NullTime{Time: dbtime.Now(), Valid: true},
```

Comment:
Using api.Clock to allow mocking time in tests 🙂 

```suggestion
		LastInvalidatedAt: sql.NullTime{Time: api.Clock.Now(), Valid: true},
```